src/valid/type.rs

```diff
@@ -328,19 +328,21 @@ impl super::Validator {
                     return Err(TypeError::InvalidArrayBaseType(base));
                 }
 
-                //Note: `unwrap()` is fine, since `Layouter` goes first and calls it
-                let base_size = types[base].inner.size(constants);
-                if stride < base_size {
-                    return Err(TypeError::InsufficientArrayStride { stride, base_size });
+                let base_layout = self.layouter[base];
+                let expected_stride = base_layout.to_stride();
+                if stride != expected_stride {
+                    return Err(TypeError::InvalidArrayStride {
+                        stride,
+                        expected: expected_stride,
+                    });
                 }
 
-                let general_alignment = self.layouter[base].alignment;
+                let general_alignment_mask = base_layout.alignment.get() - 1;
                 let uniform_layout = match base_info.uniform_layout {
                     Ok(base_alignment) => {
                         // combine the alignment requirements
-                        let align = ((base_alignment.unwrap().get() - 1)
-                            | (general_alignment.get() - 1))
-                            + 1;
+                        let align =
+                            ((base_alignment.unwrap().get() - 1) | general_alignment_mask) + 1;
```

Comment:
Wouldn't it be simpler and clearer just to use `max`?

Reply:
whoa, nicely spotted!